sway-fmt-v2/src/utils/ty.rs

```diff
@@ -36,34 +35,38 @@ fn format_infer(
     Ok(())
 }
 
+/// Currently does not apply formatting, just pushes the str version of span
 fn format_array(
-    _formatted_code: &mut FormattedCode,
-    _formatter: &mut Formatter,
-    _underscore_token: SquareBrackets<TyArrayDescriptor>,
+    formatted_code: &mut FormattedCode,
+    array_descriptor: SquareBrackets<TyArrayDescriptor>,
 ) -> Result<(), FormatterError> {
-    todo!()
+    write!(formatted_code, "{}", array_descriptor.span().as_str())?;
```

Comment:
Same here and a few other places, we don't need the `write` macro if we aren't joining anything I think

Reply:
One benefit of using `write!` is that it gives us the flexibility of easily refactoring the kind of buffer that we write into as `write!` supports writing into any type `T` where `T: std::fmt::Write`. I think it's fine to use `write!` generally, perhaps even preferred.